src/main.rs

```diff
@@ -304,32 +302,48 @@ fn main() {
 	});
 
 	window.set_root(vertical_container);
-	// kick off a thread that will check for an update in the background
-	let update_available_clone = update_available.clone();
-	let update_check_done_clone = update_check_done.clone();
-	let update_checker = std::thread::spawn(move || {
-		update_available_clone.store(check_for_updates(), Ordering::SeqCst);
-		update_check_done_clone.store(true, Ordering::SeqCst);
-	});
-	let update_check_done_clone = update_check_done;
-	let help_screen_clone = help_screen;
+
+	let update_checker_join_handle = {
+		let updates = &mut config.lock().unwrap().updates;
+		let config = config.clone();
+		let update_available = update_available.clone();
+		let update_check_done = update_check_done.clone();
+
```

Comment:
Your version is a lot nicer than what I did initially.

Reply:
Thanks!